src/EditorFeatures/CSharpTest/Diagnostics/UseImplicitOrExplicitType/UseImplicitTypeTests.cs

```diff
@@ -1441,5 +1442,24 @@ static void M()
                 diagnosticId: IDEDiagnosticIds.UseImplicitTypeDiagnosticId,
                 diagnosticSeverity: DiagnosticSeverity.Error);
         }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsUseImplicitType), Trait(Traits.Feature, Test.Utilities.Traits.Features.Tuples)]
+        public async Task SuggestVarOnLocalWithIntrinsicTypeTuple()
+        {
+            await TestAsync(
+@"class C { static void M() { [|(int a, string)|] s = (a: 1, ""hello""); } }",
+@"class C { static void M() { var s = (a: 1, ""hello""); } }", 
+options: ImplicitTypeEverywhere(),
+parseOptions: TestOptions.Regular.WithTuplesFeature());
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsUseImplicitType), Trait(Traits.Feature, Test.Utilities.Traits.Features.Tuples)]
+        public async Task TupleWithDifferentNames()
+        {
+            await TestMissingAsync(
+@"class C { static void M() { [|(int, string)|] s = (c: 1, d: ""hello""); } }",
+options: ImplicitTypeEverywhere(),
+parseOptions: TestOptions.Regular.WithTuplesFeature());
+        }
```

Comment:
This is a feature specific comment. This feature has an option that suggests var only in cases where the type is apparent from RHS of the assignment. Currently, it would suggest var for something like `Tuple.Create(1, true)`. Naturally, it should now support `(int, bool) s = (1, true)` cases. 

It could be just a matter of adding a test and making sure it works, or the feature may need some tweaking to get that case working. Eitherway, feel free to add a test and a workitem if further work is required.